### PR TITLE
Add buffer types to each message related to the planner

### PR DESCRIPTION
**Pull request by abreit001**
Added buffer types to the planner messages, drive arc messages, and goal and pose messages so they can correctly use CCSDS protocol across apps.

## How has this been tested?
Built and ran the planner app and associated unit tests.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have read and followed the `CONTRIBUTING.md` file
- [ ] I have applied the .clang-format file to my changes
- [x] I have commented my code using Doxygen style comments (see Contributing MD for examples), particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the design documentation 
- [ ] I have updated any relevant parameters in the internal ICD
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If this is a bug-fix, I have added tests that can catch similar bugs in the future
- [x] New and existing unit tests pass locally with my changes
- [ ] For cfe apps I have reserved an empty address for my msgs_ids and perf_ids from the [tracking list](https://docs.google.com/spreadsheets/d/1yRUhNTRBOiGLswlsWftNaYlwGKDeNQ_fCnDYFrc_7fU/edit#gid=772812381) and checked that it does not clash with other apps
- [ ] For new added message types, there is a timestamp included via either an explicit timestamp variable in the struct or buried in the a tlmHeader.

### CODE REVIEW

**Review by bkolligs — DISMISSED** [review]
LGTM

File: fsw/public_inc/drive_arc_msg.h [hunk 1]
```diff
@@ -23,6 +23,7 @@
  * Type definition (MOONRANGER drive arc packet)
  */
 typedef struct {
+  uint8 TlmHeader[CFE_SB_TLM_HDR_SIZE];
```
Comment (tenzincrouch) [inline fsw/public_inc/drive_arc_msg.h:26]: This should only be in the below struct to my understanding.

**Review by tenzincrouch — CHANGES_REQUESTED** [review]
One minor thing to address.

**Review by tenzincrouch — APPROVED** [review]
LGTM

**Pull request merged**